File: 380.cpp
class RandomizedSet {
public:
	unordered_map<int,int> map; 
	vector<int> l; 
    bool insert(int val) {
		if (map.count(val)) return false;  
		map[val] = l.size();  
        l.push_back(val); 
		return true; 
    }
    
    bool remove(int val) {
		if (map.count(val) == 0) return false;
        int swapElem = l[l.size()-1]; 
        l[l.size()-1] = val; 
        l[map[val]] = swapElem; 
        map[swapElem] = map[val]; 
		map.erase(val);
        l.pop_back();
		return true; 
    }
    
    int getRandom() {
		return l[rand() % l.size()]; 
    }
};

File: 380.py
class RandomizedSet:

    def __init__(self):
        self.elements = {}
        self.n = 0

    def insert(self, val: int) -> bool:
        if val not in self.elements:
            self.elements[val] = True
            self.n += 1
            return True
        return False

    def remove(self, val: int) -> bool:
        if val in self.elements:
            self.elements.pop(val)
            self.n -= 1
            return True
        return False

    def getRandom(self) -> int:
        rand = random.randint(0, self.n-1)
        return list(self.elements.keys())[rand]

